### PR TITLE
Add liveness/readiness probes for nsmd and vppagent-dataplane

### PR DESCRIPTION
**Pull request by rdimitrov**
## Description

Add liveness/readiness probes for nsmd and vppagent-dataplane. The changes consist of:
* Created `/liveness` and `/readiness` HTTP handlers for both components
* Each component sets specific flags stating if its dependencies initialized okay or not
* Updated the `kube-testing` description files to take into account these changes
* Updated the Kubernetes configuration files to enable liveness/readiness probing

## Motivation and Context

Related to #711

## How Has This Been Tested?

- [X] Covered by existing integration testing
- [x] Added integration testing to cover
- [x] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by nickolaev** [comment]
@rdimitrov I've added a couple of checks in the PR description. It is somehow covered by the existing tests, it;s a new feature and we'll need to add something to the docs to reflect that this check is already something.

File: k8s/conf/vppagent-dataplane.yaml [hunk 1]
```diff
@@ -23,6 +23,18 @@ spec:
...
+            httpGet:
+              path: /liveness
+              port: 5555
+            initialDelaySeconds: 3
+            periodSeconds: 3
+          readinessProbe:
+            httpGet:
+              path: /readiness
+              port: 5555
+            initialDelaySeconds: 5
```
Comment (edwarnicke) [inline k8s/conf/vppagent-dataplane.yaml]: Question... what motivated this choice of initial delay?

**Comment by edwarnicke** [comment]
@rdimitrov Overall approach is nicer than what I had suggested, good job!

I am curious about the choice of initialDelay values :)

**Comment by rdimitrov** [comment]
@edwarnicke Thank you, Ed! I really appreciate it! 👍 

And related to the values, I set them more or less as a starting point. If it would've failed, I was going to increase it a bit. 
I can do some basic measurement to get a rough idea of what does it take for each component to start. And based on that, experiment a bit to see if it makes sense to tune them better.

**Comment by rdimitrov** [comment]
Updates:
* Created a dedicated file in the `docs/spec/` folder for documenting that feature
* Added measuring of the boot time for both components (printed in logging=debug). Useful for debugging purposes, for example, if a liveness probe fails due to slow starting time.
* Adjusted the timings, so the system is more stable. For example, the default value of `timeoutSeconds` is 1s was not enough as it was causing the liveness probe to trigger a fail and restart the container.

**Pull request merged**